src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosDriverFactory.scala

```diff
@@ -45,4 +57,30 @@ class MesosDriverFactory(val mesosScheduler: Scheduler, val frameworkInfo: Frame
     mesosDriver.get.stop(true)
     mesosDriver = None
   }
+
+
+  /**
+   * Create the optional credentials instance, used to authenticate calls from Chronos to Mesos.
+   */
+  def buildMesosCredentials(principal: String, secretFile: Option[String]): Credential = {
+
+    val credentialBuilder = Credential.newBuilder()
+      .setPrincipal(principal)
+
+    secretFile foreach { file =>
+      try {
+        val secretBytes = ByteString.readFrom(new FileInputStream(file))
```

Comment:
Can we have a more descriptive warning message here, i.e. "Secret file allows write permissions when it should be read only, continuing anyway"